vhost-device-vsock/src/vsock_conn.rs

```diff
@@ -174,11 +174,19 @@ impl<S: AsRawFd + Read + Write> VsockConnection<S> {
                         pkt.set_op(VSOCK_OP_RW).set_len(read_cnt as u32);
 
                         // Re-register the stream file descriptor for read and write events
-                        VhostUserVsockThread::epoll_register(
+                        if VhostUserVsockThread::epoll_modify(
                             self.epoll_fd,
                             self.stream.as_raw_fd(),
                             epoll::Events::EPOLLIN | epoll::Events::EPOLLOUT,
-                        )?;
+                        )
+                        .is_err()
+                        {
+                            VhostUserVsockThread::epoll_register(
```

Comment:
Okay, but we already copied the data in the buffer, so we should do update the `rx_cnt`  and `last_fwd_cnt ` also if we fail while calling `epoll_register` (pre-existing issue), right?

Reply:
mmm, in `process_rx_queue` if `self.thread_backend.recv_pkt()` fails, we discard the packet, but here we already consumed the bytes in the stream coming from UDS, so we are loosing data if `epoll_register` fails, and I'm not sure if it is the right thing to do (as you pointed out).

Reply:
I'd suggest logging this error and maybe ignore it for now. For a long term, I think we should move out this code from this function, and let's the vsock_thread handle this case properly.

Reply:
> I agree with that. Done. It would be great if we could refactor this in the context of #323

Yep, please add a comment there.

> 
> QQ: why does it continuously unregister and register epoll listener?

No idea, IIRC we unregister only when receiving RST from the guest, so I was expecting registering it only the first time. We need to investigate more this code.

Reply:
Okay, now reading `send_pkt()` maybe it came back to me, IIRC we remove it when we don't have space in the device to send packets to the guest.

Reply:
> Okay, now reading `send_pkt()` maybe it came back to me, IIRC we remove it when we don't have space in the device to send packets to the guest.

mmm, no I don't think that's the case, IIUC we centralized events with RxQueue, so as soon as it received an event from the backend, it would queue an RxOps and remove the fd from the epoll, and then add it back after reading it. This should improve the batching, but we may need to improve this to avoid to many system call.